docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -0,0 +1,108 @@
+
+Here you will find more advanced configuration options, and some practices which address the needs of running Nuclio in production environments. 
+
+
+#### In this document
+
+- [Preferred installation method](#preferred-installation-method)
+- [Multi Tenancy](#multi-tenancy)
+- [Dark site operation](#dark-site-operation)
+- [Using Kaniko as an image builder](#using-kaniko-as-an-image-builder)
+
+## Preferred installation method
+
+Even though you may, of course, use any of the available methods, we find that heavy-lifting over k8s is best done with helm charts.
+Nuclio's [helm chart](/hack/k8s/helm/nuclio/) is our preferred mode of deploying Nuclio in [Iguazio](https://www.iguazio.com/) as of writing this document.
+This is also where you'll find most of the production oriented features appearing first, and is the most tightly maintained form of deployment.
+
+Below is a quick example of how to setup the latest stable version of nuclio via helm charts:
+
+- Create a namespace for your functions, and a secret with your intended registry's credentials
+    ```sh
+    kubectl create namespace nuclio
+    ```
+    ```sh
+    read -s mypassword
+    <enter your password>
+    
+    kubectl create secret docker-registry registry-credentials --namespace nuclio \
+        --docker-username <username> \
+        --docker-password $mypassword \
+        --docker-server <URL> \
+        --docker-email ignored@nuclio.io
+    
+    unset mypassword
+    ```
+ - Checkout the nuclio project and install nuclio from its helm chart: 
+    ```sh
+    git checkout https://github.com/nuclio/nuclio.git
+    
+    helm install \
+        --set registry.secretName=registry-credentials \
+        --set registry.pushPullUrl=<your registry URL> \
+        --set controller.image.tag=latest-amd64 \
+        --set dashboard.image.tag=latest-amd64\
+        .
+    ```
+  
+  See [the helm chart's values file](/hack/k8s/helm/nuclio/values.yaml) for a full list of configurable parameters
+
+## Multi Tenancy
+
+Implementation of multi-tenancy can be done in many different forms and to various degrees. Our experience have led us to adopt the k8s approach of tenant isolation using namespaces.
+- To achieve tenant separation for various nuclio projects and functions, and to avoid cross-tenant contamination and resource races, we have opted to deploy in each namespace a fully functioning Nuclio deployment, and configure Nuclio's controller to be namespaced.
+  This means the controller will handle Nuclio resources (functions, function-events, projects) only within its own namespace. This is supported via `Values.controller.namespace` in the helm chart values.  
+- To provide ample separation on the docker image level, we highly recommend that the Nuclio deployments of various tenants will not share docker registries, or will not share a tenant, if a multi-tenant registry is used (like `docker.io` or `quay.io`) 
+ 
+## Dark site operation
```

Comment:
`Dark site` -> `Air gapped` (more commonly used). You also switch back and forth between the two. I would pick one (air gapped) and put Dark Site as an alias in the title (e.g. `Air gapped (dark site) operation` in the title, but only use air gapped in the contents